mlir/lib/Analysis/Presburger/PresburgerRelation.cpp

```diff
@@ -1015,6 +1022,17 @@ bool PresburgerRelation::hasOnlyDivLocals() const {
   });
 }
 
+PresburgerRelation PresburgerRelation::simplify() const {
+  PresburgerRelation origin = *this;
```

Comment:
Why do we make a copy here? Can we use &origin = *this?

Reply:
The simplify function is currently a const member function, and now the simplify function of IntegerRelation is a non-const member function, in order to be able to call its simplify so there is a copy, here how to change will be more appropriate?

Reply:
Makes sense, its fine for now